packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -0,0 +1,142 @@
+"""Utilizes command line args to create a Measurement Plug-In Client using template files."""
+
+import pathlib
+from typing import Any, Dict, Optional
+
+import black
+import click
+import grpc
+from mako.template import Template
+from ni_measurement_plugin_sdk_service._internal.stubs.ni.measurementlink.measurement.v2 import (
+    measurement_service_pb2 as v2_measurement_service_pb2,
+)
+from ni_measurement_plugin_sdk_service.discovery import DiscoveryClient
+from ni_measurement_plugin_sdk_service.grpc.channelpool import GrpcChannelPool
+
+from ni_measurement_plugin_sdk_generator.client._support import (
+    get_configuration_metadata_by_index,
+    get_configuration_parameters_with_type_and_default_values,
+    get_measurement_service_stub,
+    get_output_metadata_by_index,
+    get_output_parameters_with_type,
+    get_python_module_name,
+    get_python_class_name,
+    ImportType,
+)
+
+
+def _render_template(template_name: str, **template_args: Any) -> bytes:
+    file_path = str(pathlib.Path(__file__).parent / "templates" / template_name)
+    template = Template(filename=file_path, input_encoding="utf-8", output_encoding="utf-8")
+
+    try:
+        return template.render(**template_args)
+    except Exception:
+        raise click.ClickException(f'An error occurred while rendering template "{template_name}".')
+
+
+def _create_file(
+    template_name: str, file_name: str, directory_out: pathlib.Path, **template_args: Any
+) -> None:
+    output_file = directory_out / file_name
+
+    output = _render_template(template_name, **template_args).decode("utf-8")
+    formatted_output = black.format_str(
+        src_contents=output,
+        mode=black.Mode(line_length=100),
+    )
+
+    with output_file.open("w") as file:
+        file.write(formatted_output)
+
+
+@click.command()
+@click.option(
+    "-m",
+    "--module-name",
+    help="Name for the Python Measurement Plug-In Client module.",
+)
+@click.option(
+    "-c",
+    "--class-name",
+    help="Name for the Python Measurement Plug-In Client Class in the generated module.",
+)
+@click.option(
+    "-s",
+    "--measurement-service-class",
+    help="The service class of your measurement.",
+)
+@click.option(
+    "-o",
+    "--directory-out",
+    help="Output directory for Measurement Plug-In Client files. Default: '<current_directory>/<module_name>'",
+)
+def create_client(
+    module_name: str,
+    class_name: str,
+    measurement_service_class: str,
+    directory_out: Optional[str],
+) -> None:
+    """Generates a Python Measurement Plug-In Client module for the measurement service.
+
+    You can use the generated module to interact with the corresponding measurement service.
+    """
+    channel_pool = GrpcChannelPool()
+    discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
+    import_modules: Dict[str, ImportType] = {}
```

Comment:
Since this is iterated twice in the mako and based on how it is created. Can we be two separate lists, one for built-in types and the other for custom types?

Reply:
Sure, updated the dictionaries with two separate sets.